src/main/java/seedu/pill/util/Parser.java

```diff
@@ -0,0 +1,365 @@
+     */
+    public void parseCommand(String input) {
+        try {
+            String[] splitInput = input.split("\\s+");
+            String commandString = splitInput[0].toLowerCase();
+            String argument = splitInput.length > 1 ? splitInput[1] : null;
+            String flagStr = splitInput.length > 2 ? splitInput[2] : "";
+            String arguments = String.join(" ", Arrays.copyOfRange(splitInput, 1, splitInput.length));
+
+            switch (commandString) {
+            case "exit":
+                this.exitFlag = true;
+                break;
+            case "add":
+                parseAddItemCommand(arguments).execute(this.items, this.storage);
+                break;
+            case "delete":
+                parseDeleteItemCommand(arguments).execute(this.items, this.storage);
+                break;
+            case "edit":
+                parseEditItemCommand(arguments).execute(this.items, this.storage);
+                break;
+            case "find":
+                new FindCommand(arguments).execute(this.items, this.storage);
+                break;
+            case "help":
+                boolean flag = flagStr.equals("-v");
+                new HelpCommand(argument, flag).execute(this.items, this.storage);
+                break;
+            case "list":
+                if (splitInput.length > 1) {
+                    throw new PillException(ExceptionMessages.TOO_MANY_ARGUMENTS);
+                }
+                new ListCommand().execute(this.items, this.storage);
+                break;
+            case "stock-check":
+                if (splitInput.length > 2) {
+                    throw new PillException(ExceptionMessages.TOO_MANY_ARGUMENTS);
+                }
+                new StockCheckCommand(argument).execute(this.items, this.storage);
+                break;
+            default:
+                throw new PillException(ExceptionMessages.INVALID_COMMAND);
+            }
+        } catch (PillException e) {
+            PillException.printException(e);
+        }
+    }
+
+    /**
+     * Parses the user input and creates an {@code AddItemCommand} object.
+     * The input is expected to contain the item name, quantity, and optional expiry date.
+     * If a valid date is found, it must be the last element in the input.
+     * Only one date and one quantity are allowed.
+     *
+     * The method loops through the input array to determine the item name, quantity, and expiry date,
+     * applying default values when necessary (e.g., quantity defaults to 1 if not specified).
+     *
+     * @param arguments A string representing the user's input for adding an item
+     * @return An {@code AddItemCommand} containing the parsed item name, quantity, and optional expiry date.
+     * @throws PillException If the input format is invalid
+     */
+    private AddItemCommand parseAddItemCommand(String arguments) throws PillException {
```

Comment:
Maybe you can split this method up to multiple smaller methods instead of it all being in this long method